checks/evaluation/maintained.go

```diff
@@ -30,69 +33,72 @@ const (
 )
 
 // Maintained applies the score policy for the Maintained check.
-func Maintained(name string, dl checker.DetailLogger, r *checker.MaintainedData) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
-		return checker.CreateRuntimeErrorResult(name, e)
+func Maintained(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
+) checker.CheckResult {
+	// We have 7 unique probes, each should have a finding.
+	expectedProbes := []string{
+		archived.Probe,
+		activityOnIssuesByCollaboratorsMembersOrOwnersInLast90Days.Probe,
+		commitsInLast90Days.Probe,
+		wasCreatedInLast90Days.Probe,
 	}
 
-	if r.ArchivedStatus.Status {
-		return checker.CreateMinScoreResult(name, "repo is marked as archived")
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
-	// If not explicitly marked archived, look for activity in past `lookBackDays`.
-	threshold := time.Now().AddDate(0 /*years*/, 0 /*months*/, -1*lookBackDays /*days*/)
-	commitsWithinThreshold := 0
-	for i := range r.DefaultBranchCommits {
-		if r.DefaultBranchCommits[i].CommittedDate.After(threshold) {
-			commitsWithinThreshold++
-		}
+	if projectIsArchived(findings) {
+		checker.LogFindings(nonNegativeFindings(findings), dl)
```

Comment:
wouldn't we want the negative finding in this case? since we want the details to include the fact that the project is archived?

Reply:
```console
go run main.go --repo golang/mock --checks Maintained --show-details --format json | jq
{
  "date": "2023-09-27T10:11:02-07:00",
  "repo": {
    "name": "github.com/golang/mock",
    "commit": "HEAD"
  },
  "scorecard": {
    "version": "",
    "commit": "unknown"
  },
  "score": 0,
  "checks": [
    {
      "details": [
        "Info: Repository was not created in last 90 days."
      ],
      "score": 0,
      "reason": "project is archived",
      "name": "Maintained",
```

Reply:
https://github.com/ossf/scorecard/pull/3507/commits/87238fc48051021e93c41b38d73913571885be21